mipengine/udfgen/udfgen_DTOs.py

```diff
@@ -0,0 +1,150 @@
+from string import Template
+from typing import List
+from typing import Optional
+
+from mipengine.node_tasks_DTOs import TableInfo
+
+
+class UDFGenResult:
+    pass
+
+
+class TableUDFGenResult(UDFGenResult):
+    tablename_placeholder: str
+    drop_query: Template
+    create_query: Template
```

Comment:
The declarations are used only for type annotations/recommendations on mismatching types.

Pydantic cannot be used because it doesn't support the `Template` type.

Reply:
pydantic supports any type, it just needs to be configured as such
```
class UDFGenResult(BaseModel):
    class Config:
        arbitrary_types_allowed = True


class TableUDFGenResult(UDFGenResult):
    tablename_placeholder: str
    drop_query: Template
    create_query: Template

    def __eq__(self, other):
        if self.tablename_placeholder != other.tablename_placeholder:
            return False
        if self.drop_query.template != other.drop_query.template:
            return False
        if self.create_query.template != other.create_query.template:
            return False
        return True

    def __repr__(self):
        return (
            f"TableUDFGenResult("
            f"tablename_placeholder='{self.tablename_placeholder}', "
            f"drop_query='{self.drop_query.template}', "
            f"create_query='{self.create_query.template}'"
            f")"
        )
```

Reply:
Nice! Didn't know that.
Fixed.